apiserver/facades/agent/secretsmanager/access.go

```diff
@@ -22,7 +23,14 @@ func secretOwner(agentAppName string) common.GetAuthFunc {
 	}
 }
 
-func (s *SecretsManagerAPI) checkCanRead(uri *coresecrets.URI) error {
-	// TODO(wallyworld)
-	return nil
+type canReadSecretFunc func(consumer names.Tag, uri *coresecrets.URI) bool
+
+func canReadSecret(authorizer facade.Authorizer) canReadSecretFunc {
+	return func(consumer names.Tag, uri *coresecrets.URI) bool {
+		if !authorizer.AuthOwner(consumer) {
+			return false
+		}
+		// TODO(wallyworld)
```

Comment:
Do we have anything else want to check or we should just remove it?

Reply:
still have to handle ACLs